clippy_utils/src/hir_utils.rs

```diff
@@ -65,7 +66,9 @@ impl<'a, 'tcx> SpanlessEq<'a, 'tcx> {
     }
 
     pub fn eq_block(&mut self, left: &Block<'_>, right: &Block<'_>) -> bool {
-        self.inter_expr().eq_block(left, right)
+        !self.cannot_be_compared_block(left)
```

Comment:
This would be better on `eq_block` from `HirEqInterExpr`. That way it gets checked on every block seen.

Reply:
I done in https://github.com/rust-lang/rust-clippy/pull/9006/commits/39ffda014debd9d1d4126a8996d39af934fe8d94